---
title: "Snodgrass_EEOB546_R_Assignment"
author: "Samantha Snodgrass"
date: "10/9/2018"
output: html_document
---
Load libraries
```{r}
library(tidyverse)
```
#Part 1: Formatting the data
##Data Inspection
Load and inspect `fang_et_al_genotypes.txt` and `snp_position.txt` for their structure and dimensions
Read in text files as tibbles
```{r}
fang_et_al_genotypes <- read_tsv('fang_et_al_genotypes.txt') 
snp_position <- read_tsv('snp_position.txt')
```

Inspect the structure and dimensions of data files
```{r}
str(fang_et_al_genotypes)
str(snp_position) # Notice that the chromosome and position variables are chr, not integer --> likely due to multiple and unknown positions
#attributes(fang_et_al_genotypes) #column and row names, not particularly helpful with this file
attributes(snp_position)
```

##Data processing
Goal: For maize (Group = ZMMIL, ZMMLR, ZMMMR) and teosinte (Group = ZMPBA, ZMPIL, ZMPJA), create:
  - 10 files (1/chromosome) with SNPs ordered by increasing position values, missing data = ?
  - 10 files (1/chromosome) with SNPs ordered based on decreasing position values, missing data = -

Remove the multiple and unknown SNPs

```{r}
#View(snp_position) #Use toggles to be sure what string is used for multiple and unknown markers
filtered.snp.position <- filter(snp_position, Chromosome != 'unknown', Chromosome != 'multiple', Position != 'multiple')
filtered.snp.position$Chromosome <- as.integer(filtered.snp.position$Chromosome) #changes chr to int
filtered.snp.position$Position <- as.integer(filtered.snp.position$Position)
str(filtered.snp.position)
View(filtered.snp.position) #check to make sure all multiple and unknown markers were removed
```

Parse out the maize and teosinte samples by group
```{r}
maize.genotypes <- filter(fang_et_al_genotypes, Group %in% c('ZMMIL', 'ZMMLR', 'ZMMMR'))
teosinte.genotypes <- filter(fang_et_al_genotypes, Group %in% c('ZMPBA', 'ZMPIL', 'ZMPJA'))
```

Join the two files
-Will require the transpose command
-Double check to make sure that it hasn't coerced the data into a new type, may require changing the current dataframes into another datastructure first and then transposing and then reverting back to a dataframe
```{r}
maize.transposed.genotypes <- t(maize.genotypes) #transposes tibble into matrix
colnames(maize.transposed.genotypes) <- maize.genotypes$Sample_ID #renames columns to sampleID
maize.transposed.genotypes <- as_tibble(maize.transposed.genotypes, rownames = NA) #matrix --> tibble while keeping row names

teosinte.transposed.genotypes <- t(teosinte.genotypes)
colnames(teosinte.transposed.genotypes) <- teosinte.genotypes$Sample_ID
teosinte.transposed.genotypes <- as_tibble(teosinte.transposed.genotypes, rownames = NA)

#str(maize.transposed.genotypes) #to double check the data structure of new transposed files
#str(teosinte.transposed.genotypes)
```

Use the merge command to join the dataframes. by = indicate which column to join on, all = F indicates that only things found in both datasets should be kept; colnames renames the blank header space with new identifier
```{r}
all.maize <- merge(maize.transposed.genotypes, filtered.snp.position, by.x = 0, by.y = 'SNP_ID', all = FALSE)
colnames(all.maize)[1] <- "SNP_ID"
all.teosinte <- merge(teosinte.transposed.genotypes, filtered.snp.position, by.x = 0, by.y = 'SNP_ID', all = FALSE)
colnames(all.teosinte)[1] <- "SNP_ID"
```

Parse both all maize and teosinte dataframes by chromosome number
```{r}
maize.Chr1 <- filter(all.maize, Chromosome == 1)
maize.Chr2 <- filter(all.maize, Chromosome == 2)
maize.Chr3 <- filter(all.maize, Chromosome == 3)
maize.Chr4 <- filter(all.maize, Chromosome == 4)
maize.Chr5 <- filter(all.maize, Chromosome == 5)
maize.Chr6 <- filter(all.maize, Chromosome == 6)
maize.Chr7 <- filter(all.maize, Chromosome == 7)
maize.Chr8 <- filter(all.maize, Chromosome == 8)
maize.Chr9 <- filter(all.maize, Chromosome == 9)
maize.Chr10 <- filter(all.maize, Chromosome == 10)
teosinte.Chr1 <- filter(all.teosinte, Chromosome == 1)
teosinte.Chr2 <- filter(all.teosinte, Chromosome == 2)
teosinte.Chr3 <- filter(all.teosinte, Chromosome == 3)
teosinte.Chr4 <- filter(all.teosinte, Chromosome == 4)
teosinte.Chr5 <- filter(all.teosinte, Chromosome == 5)
teosinte.Chr6 <- filter(all.teosinte, Chromosome == 6)
teosinte.Chr7 <- filter(all.teosinte, Chromosome == 7)
teosinte.Chr8 <- filter(all.teosinte, Chromosome == 8)
teosinte.Chr9 <- filter(all.teosinte, Chromosome == 9)
teosinte.Chr10 <- filter(all.teosinte, Chromosome == 10)
```

Sort by position, generating output files in both directions
-change the missing data symbol in the decreasing position files
```{r}
maize.Chr1.ascend <- arrange(maize.Chr1, maize.Chr1$Position)
maize.Chr1.descend <- arrange(maize.Chr1, desc(maize.Chr1$Position))
maize.Chr2.ascend <- arrange(maize.Chr2, maize.Chr2$Position)
maize.Chr2.descend <- arrange(maize.Chr2, desc(maize.Chr2$Position))
maize.Chr3.ascend <- arrange(maize.Chr3, maize.Chr3$Position)
maize.Chr3.descend <- arrange(maize.Chr3, desc(maize.Chr3$Position))
maize.Chr4.ascend <- arrange(maize.Chr4, maize.Chr4$Position)
maize.Chr4.descend <- arrange(maize.Chr4, desc(maize.Chr4$Position))
maize.Chr5.ascend <- arrange(maize.Chr5, maize.Chr5$Position)
maize.Chr5.descend <- arrange(maize.Chr5, desc(maize.Chr5$Position))
maize.Chr6.ascend <- arrange(maize.Chr6, maize.Chr6$Position)
maize.Chr6.descend <- arrange(maize.Chr6, desc(maize.Chr6$Position))
maize.Chr7.ascend <- arrange(maize.Chr7, maize.Chr7$Position)
maize.Chr7.descend <- arrange(maize.Chr7, desc(maize.Chr7$Position))
maize.Chr8.ascend <- arrange(maize.Chr8, maize.Chr8$Position)
maize.Chr8.descend <- arrange(maize.Chr8, desc(maize.Chr8$Position))
maize.Chr9.ascend <- arrange(maize.Chr9, maize.Chr9$Position)
maize.Chr9.descend <- arrange(maize.Chr9, desc(maize.Chr9$Position))
maize.Chr10.ascend <- arrange(maize.Chr10, maize.Chr10$Position)
maize.Chr10.descend <- arrange(maize.Chr10, desc(maize.Chr10$Position))

teosinte.Chr1.ascend <- arrange(teosinte.Chr1, teosinte.Chr1$Position)
teosinte.Chr1.descend <- arrange(teosinte.Chr1, desc(teosinte.Chr1$Position))
teosinte.Chr2.ascend <- arrange(teosinte.Chr2, teosinte.Chr2$Position)
teosinte.Chr2.descend <- arrange(teosinte.Chr2, desc(teosinte.Chr2$Position))
teosinte.Chr3.ascend <- arrange(teosinte.Chr3, teosinte.Chr3$Position)
teosinte.Chr3.descend <- arrange(teosinte.Chr3, desc(teosinte.Chr3$Position))
teosinte.Chr4.ascend <- arrange(teosinte.Chr4, teosinte.Chr4$Position)
teosinte.Chr4.descend <- arrange(teosinte.Chr4, desc(teosinte.Chr4$Position))
teosinte.Chr5.ascend <- arrange(teosinte.Chr5, teosinte.Chr5$Position)
teosinte.Chr5.descend <- arrange(teosinte.Chr5, desc(teosinte.Chr5$Position))
teosinte.Chr6.ascend <- arrange(teosinte.Chr6, teosinte.Chr6$Position)
teosinte.Chr6.descend <- arrange(teosinte.Chr6, desc(teosinte.Chr6$Position))
teosinte.Chr7.ascend <- arrange(teosinte.Chr7, teosinte.Chr7$Position)
teosinte.Chr7.descend <- arrange(teosinte.Chr7, desc(teosinte.Chr7$Position))
teosinte.Chr8.ascend <- arrange(teosinte.Chr8, teosinte.Chr8$Position)
teosinte.Chr8.descend <- arrange(teosinte.Chr8, desc(teosinte.Chr8$Position))
teosinte.Chr9.ascend <- arrange(teosinte.Chr9, teosinte.Chr9$Position)
teosinte.Chr9.descend <- arrange(teosinte.Chr9, desc(teosinte.Chr9$Position))
teosinte.Chr10.ascend <- arrange(teosinte.Chr10, teosinte.Chr10$Position)
teosinte.Chr10.descend <- arrange(teosinte.Chr10, desc(teosinte.Chr10$Position))
```

Replace the "?" with "-"
```{r}
maize.Chr1.descend[maize.Chr1.descend=="?/?"] <- "-/-"
maize.Chr2.descend[maize.Chr2.descend=="?/?"] <- "-/-"
maize.Chr3.descend[maize.Chr3.descend=="?/?"] <- "-/-"
maize.Chr4.descend[maize.Chr4.descend=="?/?"] <- "-/-"
maize.Chr5.descend[maize.Chr5.descend=="?/?"] <- "-/-"
maize.Chr6.descend[maize.Chr6.descend=="?/?"] <- "-/-"
maize.Chr7.descend[maize.Chr7.descend=="?/?"] <- "-/-"
maize.Chr8.descend[maize.Chr8.descend=="?/?"] <- "-/-"
maize.Chr9.descend[maize.Chr9.descend=="?/?"] <- "-/-"
maize.Chr10.descend[maize.Chr10.descend=="?/?"] <- "-/-"
teosinte.Chr1.descend[teosinte.Chr1.descend=="?/?"] <- "-/-"
teosinte.Chr2.descend[teosinte.Chr2.descend=="?/?"] <- "-/-"
teosinte.Chr3.descend[teosinte.Chr3.descend=="?/?"] <- "-/-"
teosinte.Chr4.descend[teosinte.Chr4.descend=="?/?"] <- "-/-"
teosinte.Chr5.descend[teosinte.Chr5.descend=="?/?"] <- "-/-"
teosinte.Chr6.descend[teosinte.Chr6.descend=="?/?"] <- "-/-"
teosinte.Chr7.descend[teosinte.Chr7.descend=="?/?"] <- "-/-"
teosinte.Chr8.descend[teosinte.Chr8.descend=="?/?"] <- "-/-"
teosinte.Chr9.descend[teosinte.Chr9.descend=="?/?"] <- "-/-"
teosinte.Chr10.descend[teosinte.Chr10.descend=="?/?"] <- "-/-"
```

Write output files
```{r}
write.csv(maize.Chr1.ascend, file = "maize.Chr1.ascend.csv", row.names = TRUE)
write.csv(maize.Chr2.ascend, file = "maize.Chr2.ascend.csv", row.names = TRUE)
write.csv(maize.Chr3.ascend, file = "maize.Chr3.ascend.csv", row.names = TRUE)
write.csv(maize.Chr4.ascend, file = "maize.Chr4.ascend.csv", row.names = TRUE)
write.csv(maize.Chr5.ascend, file = "maize.Chr5.ascend.csv", row.names = TRUE)
write.csv(maize.Chr6.ascend, file = "maize.Chr6.ascend.csv", row.names = TRUE)
write.csv(maize.Chr7.ascend, file = "maize.Chr7.ascend.csv", row.names = TRUE)
write.csv(maize.Chr8.ascend, file = "maize.Chr8.ascend.csv", row.names = TRUE)
write.csv(maize.Chr9.ascend, file = "maize.Chr9.ascend.csv", row.names = TRUE)
write.csv(maize.Chr10.ascend, file = "maize.Chr10.ascend.csv", row.names = TRUE)

write.csv(maize.Chr1.descend, file = "maize.Chr1.descend.csv", row.names = TRUE)
write.csv(maize.Chr2.descend, file = "maize.Chr2.descend.csv", row.names = TRUE)
write.csv(maize.Chr3.descend, file = "maize.Chr3.descend.csv", row.names = TRUE)
write.csv(maize.Chr4.descend, file = "maize.Chr4.descend.csv", row.names = TRUE)
write.csv(maize.Chr5.descend, file = "maize.Chr5.descend.csv", row.names = TRUE)
write.csv(maize.Chr6.descend, file = "maize.Chr6.descend.csv", row.names = TRUE)
write.csv(maize.Chr7.descend, file = "maize.Chr7.descend.csv", row.names = TRUE)
write.csv(maize.Chr8.descend, file = "maize.Chr8.descend.csv", row.names = TRUE)
write.csv(maize.Chr9.descend, file = "maize.Chr9.descend.csv", row.names = TRUE)
write.csv(maize.Chr10.descend, file = "maize.Chr10.descend.csv", row.names = TRUE)

write.csv(teosinte.Chr1.ascend, file = "teosinte.Chr1.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr2.ascend, file = "teosinte.Chr2.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr3.ascend, file = "teosinte.Chr3.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr4.ascend, file = "teosinte.Chr4.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr5.ascend, file = "teosinte.Chr5.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr6.ascend, file = "teosinte.Chr6.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr7.ascend, file = "teosinte.Chr7.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr8.ascend, file = "teosinte.Chr8.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr9.ascend, file = "teosinte.Chr9.ascend.csv", row.names = TRUE)
write.csv(teosinte.Chr10.ascend, file = "teosinte.Chr10.ascend.csv", row.names = TRUE)

write.csv(teosinte.Chr1.descend, file = "teosinte.Chr1.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr2.descend, file = "teosinte.Chr2.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr3.descend, file = "teosinte.Chr3.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr4.descend, file = "teosinte.Chr4.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr5.descend, file = "teosinte.Chr5.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr6.descend, file = "teosinte.Chr6.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr7.descend, file = "teosinte.Chr7.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr8.descend, file = "teosinte.Chr8.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr9.descend, file = "teosinte.Chr9.descend.csv", row.names = TRUE)
write.csv(teosinte.Chr10.descend, file = "teosinte.Chr10.descend.csv", row.names = TRUE)
```

#Part 2: Visualizing the data
Note that it may be easier to reshape the original data using the 'melt' command in the 'reshape2' package before attempting this part

For downstream analysis, I will want the data organized as columns for sample ID, SNP ID, Group, and Genotype. It will be easiest to get this format with all the various data from the original data files `snp_position.txt` and `fang_et_al_genotypes.txt`. The `snp_position.txt` won't need to be reshaped since I only need SNP_ID and chromosome as columns. 
```{r}
library(reshape2)
all.snps <- melt(fang_et_al_genotypes, id.vars = c("Sample_ID", "Group", "JG_OTU"), variable.name = "SNP_ID", value.name = "Genotype") #variable.name titles the measured variable colum, value.name names the value column, have to use the c() function for id.vars since there's multiple id variables
```

##SNPs per chromosome
Plot the total number of SNPs in our dataset on each chromosome
```{r}
library(ggplot2)
SNPbyChr <- ggplot(data = snp_position) + geom_bar(mapping = aes(x = Chromosome)) #with multiple and unknown positioned SNPs
SNPbyChr
filteredSNPbyChr <- ggplot(data = filtered.snp.position) + geom_bar(mapping = aes(x = Chromosome)) #without multiple and unknown positioned SNPs
filteredSNPbyChr
```
What groups contribute most of these SNPs -- Continued after I created the homozygous columns for the second question, so I'll put that down there so it'll run with the appropriate dataset

##Missing data and amount of heterozygosity

```{r}
all.snps <- mutate(all.snps, homozygous = recode(Genotype, "A/A" = "Hom", "T/T" = "Hom", "G/G" = "Hom", "C/C" = "Hom", "A/T" = "Het", "A/C" = "Het", "A/G" = "Het", "T/C" = "Het", "T/G" = "Het", "T/A" = "Het", "C/A" = "Het", "C/T" = "Het", "C/G" = "Het", "G/A" = "Het", "G/T" = "Het", "G/C" = "Het", "?/?" = "Miss")) # Het = heterozygote, Hom = homozygote, Miss = missing
all.snps.bygroup <- arrange(all.snps, all.snps$Group) #arrange data by group
all.snps.bysample <- arrange(all.snps, all.snps$Sample_ID) #arrange data by Sample_ID

HomozygositybyGroup <- ggplot(data = all.snps.bygroup) + geom_bar(mapping = aes(x = Group, color = homozygous, fill = homozygous)) + theme(axis.text.x = element_text(angle = 90, hjust = 1))
HomozygositybyGroup

HomozygositybySample <- ggplot(data = all.snps.bysample) + geom_bar(mapping = aes(x = Sample_ID, color = homozygous, fill = homozygous)) + theme(axis.text.x = element_text(angle = 90, hjust = 1))
HomozygositybySample


```

##Question 1 continued
What groups contribute most of these SNPs -- Continued
```{r}
#if (lengths(tapply(all.snps.HetHom$Genotype, all.snps.HetHom$SNP_ID, unique)) > trial) {
            #all.snps.HetHom$TrueSNP = TRUE}
  #else {all.snps.HetHom$TrueSNP = FALSE} #Didn't work

all.snps.HetHom <- filter(all.snps, all.snps$homozygous != "Miss") #removes missing genotypes
all.snps.HetHom <- mutate(all.snps.HetHom, TrueSNP = FALSE)
Filtered.ZDIPL <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZDIPL") #Filter to individual groups without missing genotypes
Filtered.ZPERR <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZPERR")
Filtered.ZLUXR <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZLUXR")
Filtered.ZMHUE <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMHUE")
Filtered.ZMPBA <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMPBA")
Filtered.ZMPJA <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMPJA")
Filtered.ZMXCH <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMXCH")
Filtered.ZMXCP <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMXCP")
Filtered.ZMXNO <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMXNO")
Filtered.ZMXNT <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMXNT")
Filtered.ZMPIL <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMPIL")
Filtered.ZMXIL <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMXIL")
Filtered.ZMMLR <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMMLR")
Filtered.ZMMMR <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMMMR")
Filtered.ZMMIL <- filter(all.snps.HetHom, all.snps.HetHom$Group == "ZMMIL")
Filtered.TRIPS <- filter(all.snps.HetHom, all.snps.HetHom$Group == "TRIPS")

tapply(Filtered.TRIPS$Genotype, Filtered.TRIPS$SNP_ID, unique) %>% lengths() -> temp #Determines the length of each element, i.e. how many unique genotypes there are by SNP_ID
merged.TRIPS <- merge(temp, Filtered.TRIPS, by.x = 0, by.y = 'SNP_ID', all = FALSE) #merges the lengths in the temp with the filtered data by group
Filtered.TRIPS$TrueSNP <- merged.TRIPS$x > 1 #adds a logical column indicated whether or not it is a true SNP for that marker for that group

tapply(Filtered.ZDIPL$Genotype, Filtered.ZDIPL$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZDIPL, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZDIPL$TrueSNP <- merged$x > 1

tapply(Filtered.ZPERR$Genotype, Filtered.ZPERR$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZPERR, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZPERR$TrueSNP <- merged$x > 1

tapply(Filtered.ZLUXR$Genotype, Filtered.ZLUXR$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZLUXR, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZLUXR$TrueSNP <- merged$x > 1

tapply(Filtered.ZMHUE$Genotype, Filtered.ZMHUE$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMHUE, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMHUE$TrueSNP <- merged$x > 1

tapply(Filtered.ZMPBA$Genotype, Filtered.ZMPBA$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMPBA, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMPBA$TrueSNP <- merged$x > 1

tapply(Filtered.ZMPJA$Genotype, Filtered.ZMPJA$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMPJA, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMPJA$TrueSNP <- merged$x > 1

tapply(Filtered.ZMXCH$Genotype, Filtered.ZMXCH$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMXCH, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMXCH$TrueSNP <- merged$x > 1

tapply(Filtered.ZMXCP$Genotype, Filtered.ZMXCP$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMXCP, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMXCP$TrueSNP <- merged$x > 1

tapply(Filtered.ZMXNO$Genotype, Filtered.ZMXNO$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMXNO, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMXNO$TrueSNP <- merged$x > 1

tapply(Filtered.ZMXNT$Genotype, Filtered.ZMXNT$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMXNT, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMXNT$TrueSNP <- merged$x > 1

tapply(Filtered.ZMPIL$Genotype, Filtered.ZMPIL$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMPIL, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMPIL$TrueSNP <- merged$x > 1

tapply(Filtered.ZMXIL$Genotype, Filtered.ZMXIL$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMXIL, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMXIL$TrueSNP <- merged$x > 1

tapply(Filtered.ZMMLR$Genotype, Filtered.ZMMLR$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMMLR, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMMLR$TrueSNP <- merged$x > 1

tapply(Filtered.ZMMMR$Genotype, Filtered.ZMMMR$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMMMR, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMMMR$TrueSNP <- merged$x > 1

tapply(Filtered.ZMMIL$Genotype, Filtered.ZMMIL$SNP_ID, unique) %>% lengths() -> temp
merged <- merge(temp, Filtered.ZMMIL, by.x = 0, by.y = 'SNP_ID', all = FALSE) 
Filtered.ZMMIL$TrueSNP <- merged$x > 1

Regrouped <- rbind.data.frame(Filtered.TRIPS, Filtered.ZDIPL) #Bringing all the data back together again for graphing purposes
Regrouped <- rbind(Regrouped, Filtered.ZLUXR, Filtered.ZMHUE, Filtered.ZMMIL, Filtered.ZMMLR, Filtered.ZMMMR, Filtered.ZMPBA,  Filtered.ZMPIL, Filtered.ZMPJA, Filtered.ZMXCH, Filtered.ZMXCP, Filtered.ZMXIL, Filtered.ZMXNO, Filtered.ZMXNT)

SNPbyGroup <- ggplot(data = Regrouped) + geom_bar(mapping = aes(x = Group, color = TrueSNP, fill = TrueSNP))+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
SNPbyGroup
```

##Visualize another feature of the dataset (Personal choice)
See if genotype has any patterns by group (for example, does one group tend to have more A/T's than another?)
```{r}
#all.snps$SNPHom <- paste(all.snps$Genotype, all.snps$homozygous)
#all.snps.bygroup <- arrange(all.snps, all.snps$Group, all.snps$SNPHom) #rerun with new column
SNPHom.plot <- ggplot(data = all.snps.bygroup) + geom_bar(mapping = aes(x = Group, color = Genotype, fill = Genotype)) + theme(axis.text.x = element_text(angle = 90, hjust = 1))
SNPHom.plot
```

